data-prepper-plugins/sns-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/sns/SnsSinkServiceTest.java

```diff
@@ -0,0 +1,155 @@
+        return new SnsSinkService(snsSinkConfig,
+                snsClient,
+                pluginMetrics, mock(PluginFactory.class),
+                mock(PluginSetting.class),
+                mock(ExpressionEvaluator.class));
+    }
+
+    @Test
+    public void sns_sink_test_with_empty_collection_records(){
+        numberOfRecordsSuccessCounter = mock(Counter.class);
+        SnsSinkService snsSinkService = createObjectUnderTest();
+        snsSinkService.output(List.of());
+        verifyNoInteractions(snsClient);
+        verifyNoInteractions(numberOfRecordsSuccessCounter);
+        verifyNoInteractions(numberOfRecordsFailedCounter);
+    }
+
+    @Test
+    public void sns_sink_test_with_single_collection_record_success_push_to_sns(){
+        when(sdkHttpResponse.statusCode()).thenReturn(500);
+        SnsSinkService snsSinkService = createObjectUnderTest();
+        final Record<Event> eventRecord = new Record<>(JacksonEvent.fromMessage("{\"message\":\"c3f847eb-333a-49c3-a4cd-54715ad1b58a\"}"));
+        Collection<Record<Event>> records = List.of(eventRecord);
+        snsSinkService.output(records);
+        verify(numberOfRecordsSuccessCounter).increment(records.size());
+    }
+
+    @Test
+    public void sns_sink_test_with_multiple_collection_record_success_push_to_sns(){
+        numberOfRecordsSuccessCounter = mock(Counter.class);
+        SnsSinkService snsSinkService = createObjectUnderTest();
+        final Record<Event> eventRecord = new Record<>(JacksonEvent.fromMessage("{\"message\":\"c3f847eb-333a-49c3-a4cd-54715ad1b58a\"}"));
+        Collection<Record<Event>> records = new ArrayList<>();
+        for(int recordSize = 0; recordSize <= 11 ; recordSize++) {
+            records.add(eventRecord);
+        }
+        snsSinkService.output(records);
+        assertThat(sdkHttpResponse.statusCode(),equalTo(200));
+        verify(numberOfRecordsSuccessCounter,times(2));
+    }
+
+    @Test
+    public void sns_sink_test_with_single_collection_record_failed_to_push_to_sns() throws IOException {
+        final Counter snsSinkObjectsEventsFailedCounter = mock(Counter.class);
+        when(pluginMetrics.counter(SnsSinkService.NUMBER_OF_RECORDS_FLUSHED_TO_SNS_FAILED)).thenReturn(snsSinkObjectsEventsFailedCounter);
+        final SnsException snsException = (SnsException)SnsException.builder().message("internal server error").awsErrorDetails(AwsErrorDetails.builder().errorMessage("internal server error").build()).build();
+        when(snsClient.publishBatch(any(PublishBatchRequest.class))).thenThrow(snsException);
+        SnsSinkService snsSinkService = createObjectUnderTest();
+        final Record<Event> eventRecord = new Record<>(JacksonEvent.fromMessage("{\"message\":\"c3f847eb-333a-49c3-a4cd-54715ad1b58a\"}"));
+        Collection<Record<Event>> records = List.of(eventRecord);
+        snsSinkService.output(records);
+        verify(snsSinkObjectsEventsFailedCounter).increment(records.size());
+    }
```

Comment:
See my comment above about verification. Please verify the calls to the SNS client using the pattern established above.

Reply:
Thanks @dlvenable for your feedback, we have incorporated the changes and available for review.